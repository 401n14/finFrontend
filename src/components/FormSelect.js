import React from 'react';

function FormSelect(props) {
    let options = [];

    options = props.list.map((item, ind)=> {
      return (
        <option key={ind} value={item.name}>
          {item.name}
        </option>
      );
    });
  
    return (
      <select
        onChange={props.onChange}
      >
          <option> -- Select Language -- </option>
        {options}
      </select>
    );
}

export default FormSelect;