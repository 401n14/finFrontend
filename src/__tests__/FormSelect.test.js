import React from 'react';
import FormSelect from '../components/FormSelect';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({adapter:new Adapter()});

let options = {
    "Languages": [
      { "name": "English" },
      { "name": "Espanol"}
    ]
  }

describe('Snapshot Test', () => {
    it('renders correctly, Snapshot', () => {
        const page = renderer.create(<FormSelect 
            list={options.Languages}
             />).toJSON();
        expect(page).toMatchSnapshot();
    });
});

describe('Form Tests', () => {
    it('renders without a user & calls the Loading component', () => {
        let wrapper = mount(<FormSelect
            list={options.Languages}
            />)
        expect(wrapper.find('option').at(0).text()).toEqual(' -- Select Language -- ');
        expect(wrapper.find('option').at(1).text()).toEqual('English');
        expect(wrapper.find('option').at(2).text()).toEqual('Espanol');

    });
});