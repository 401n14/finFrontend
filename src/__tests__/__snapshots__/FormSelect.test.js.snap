// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly, Snapshot 1`] = `
<select>
  <option>
     -- Select Language -- 
  </option>
  <option
    value="English"
  >
    English
  </option>
  <option
    value="Espanol"
  >
    Espanol
  </option>
</select>
`;
